ast/src/parsed/display.rs

```diff
@@ -1195,6 +1200,10 @@ mod tests {
                 ("let x = (1 + (|i| i + 2));", "let x = 1 + (|i| i + 2);"),
                 // Index access
                 ("(|i| i)[j];", "(|i| i)[j];"),
+                // Lambda expression
+                ("|i| (|x| x) + i;", "|i| (|x| x) + i;"),
+                ("|i| (|x| x + i);", "|i| (|x| x + i);"),
+                ("|i| |x| x + i;", "|i| (|x| x + i);"),
```

Comment:
Can you add another test set (with all combinations of parentheses) where the lambda expression is inside a binary operator, like `|i| i + |x| x`?